lib/android-helpers.js

```diff
@@ -96,23 +96,109 @@ helpers.getJavaVersion = async function () {
   return javaVer;
 };
 
+/**
+ * Installs the given certificate on a rooted real device or
+ * an emulator. The emulator must be executed with `-writable-system`
+ * command line option and adb daemon should be running in root
+ * mode for this method to work properly. The method also requires
+ * openssl tool to be available on the destination system.
+ * Read https://github.com/appium/appium/issues/10964
+ * for more details on this topic
+ *
+ * @param {ADB} adb - The ADB helper instance
+ * @param {Buffer|string} cert - base64-decoded content of the actual certificate
+ * represented as a string or a buffer
+ */
+helpers.installMitmCertificate = async function (adb, cert) {
+  try {
+    await fs.which('openssl');
+  } catch (err) {
+    throw new Error('openssl tool is required to be installed');
+  }
+
+  if (!_.isBuffer(cert)) {
+    cert = Buffer.from(cert, 'base64');
+  }
+
+  const tempCert = await tempDir.open({
+    prefix: 'cert',
+    suffix: '.cer'
+  });
+  try {
+    await fs.writeFile(tempCert.path, cert);
+    let {stdout} = await exec('openssl', ['x509', '-noout', '-hash', '-in', tempCert.path]);
+    const certHash = stdout.trim();
+    logger.debug(`Got certificate hash: ${certHash}`);
+    logger.debug('Preparing certificate content');
+    ({stdout} = await exec('openssl', ['x509', '-in', tempCert.path], {isBuffer: true}));
+    let dstCertContent = stdout;
+    ({stdout} = await exec('openssl', ['x509', '-in', tempCert.path, '-text', '-fingerprint', '-noout'], {isBuffer: true}));
+    dstCertContent = dstCertContent.concat(stdout);
+    const dstCert = await tempDir.open({
+      prefix: certHash,
+      suffix: '.0'
+    });
+    try {
+      await fs.writeFile(dstCert.path, dstCertContent);
+      logger.debug('Mounting /system endpoint');
+      await adb.shell(['mount', '-orw,remount', '/system']);
+      const dstPath = '/system/etc/security/cacerts';
+      logger.debug(`Uploading the generated certificate from '${dstCert.path}' to '${dstPath}'`);
+      await adb.push(dstCert.path, dstPath);
+      logger.debug('Remounting the remote file system');
+      await exec(adb.executable.path, [adb.executable.defaultArgs, 'remount']);
```

Comment:
I don't think we should be dependent upon internals of another module. There is the function `adb.adbExec`. If that is too heavyweight, another function should be added to just run ad-hoc commands (along the lines of `adb.createSubProcess`).